### PR TITLE
Implemented Horner's scheme for evaluating UnivariatePolynomial

### PR DESCRIPTION
**Pull request by CodeMaxx**
@isuruf @Sumith1896

### CODE REVIEW

File: symengine/polynomial.cpp [hunk 1]
```diff
@@ -468,14 +468,18 @@ Expression UnivariatePolynomial::max_coef() const {
...
+    Expression result, x_pow;
+
+    for (auto it = dict_.rbegin(); it != dict_.rend(); ++it) {
+        x_pow = pow_ex(x, last_deg - it->first);
+        last_deg = it->first;
+        result = it->second + expand(x_pow * result);
     }
-    return ans;
+    x_pow = pow_ex(x, last_deg);
+    result = expand(result * x_pow);
```
Comment (isuruf) [inline symengine/polynomial.cpp]: Why is the `expand` function called here?

Comment (CodeMaxx) [inline symengine/polynomial.cpp]: For handling a result like `1/2(31 + 4*a)`. It should be changed to  `(31/2 + 2*a)` .

**Comment by isuruf** [comment]
I think this PR should be closed, as `expand` is expensive and it is called on the coefficient itself, which is not needed

**Comment by CodeMaxx** [comment]
Closing. See the example [here](https://github.com/symengine/symengine/pull/897/files#r58304067)

**Pull request closed without merging**